### PR TITLE
Replace deprecated set-env with new format

### PR DESCRIPTION
**Pull request by gaiksaya**
*Issue #, if available:*
P40467659

*Description of changes:*
GitHub HAS ALREADY deprecate its set-env function through stdout as a security issue has been raised here:
https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/

GitHub did not issue any exact date as of now, and they just straight up disable it.
https://archive.is/PL92J

You need to change all of your set-env part to GitHub ENV files here:
https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#environment-files

*Test Results:*
https://github.com/opendistro-for-elasticsearch/opendistro-build/runs/1421029980?check_suite_focus=true#step:4:395

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/staging-build-deb.yml [hunk 1]
```diff
@@ -575,7 +579,7 @@ jobs:
           command: npx cypress run
 
   Test-AD-KIBANA:
-    needs: [sign-deb-artifacts]
+    # needs: [sign-deb-artifacts]
```
Comment (peterzhuamazon) [inline .github/workflows/staging-build-deb.yml:582]: Test and forget to remove the #?

File: .gitignore [hunk 2]
```diff
@@ -1,5 +1,8 @@
 Thumbs.db
 .DS_Store
+.gradle
+.settings
+.project
```
Comment (peterzhuamazon) [inline .gitignore:6]: Reason to make this change now?

Comment (gaiksaya) [reply .gitignore:6]: After I upgraded VS code this files came in without changing anything. so just added it here

File: .github/workflows/staging-build-deb.yml (hunk 1 above)
Comment (gaiksaya) [reply .github/workflows/staging-build-deb.yml]: Done

File: .gitignore (hunk 2 above)
Comment (gaiksaya) [reply .gitignore]: Reset to previous version. However, if anyone faces the same issue we need to modify it!

**Review by peterzhuamazon — APPROVED** [review]

**Review by sreekarjami — APPROVED** [review]

**Pull request merged**